### PR TITLE
Fix CI failures related to `cupyx.scipy.interpolate`

### PR DESCRIPTION
**Pull request by takagi**
Follows up #7206.

### CODE REVIEW

**Review by kmaehashi — APPROVED** [review]

**Comment by kmaehashi** [comment]
`cupyx_tests/scipy_tests/interpolate_tests/test_ppoly.py` needs to be fixed as well?

**Comment by takagi** [comment]
Let me run a full test here. This PR should fix the brock master CI.

File: tests/cupyx_tests/scipy_tests/interpolate_tests/test_ppoly.py [hunk 1]
```diff
@@ -22,44 +19,35 @@
...
-parametrize_cls = pytest.mark.parametrize('cls', interpolate_cls)
 
 
 @testing.with_requires("scipy")
 class TestPPolyCommon:
     """Test basic functionality for PPoly and BPoly."""
 
-    @testing.with_requires('scipy')
-    @parametrize_cls_xp
+    @pytest.mark.parametrize('cls', interpolate_cls)
```
Comment (kmaehashi) [inline tests/cupyx_tests/scipy_tests/interpolate_tests/test_ppoly.py:28]: `xp` and `scp` needs to be parameterized. Same for other tests.

**Review by kmaehashi — CHANGES_REQUESTED** [review]

File: tests/cupyx_tests/scipy_tests/interpolate_tests/test_ppoly.py (hunk 1 above)
Comment (takagi) [reply tests/cupyx_tests/scipy_tests/interpolate_tests/test_ppoly.py:28]: Fixed!

**Comment by takagi** [comment]
The CI failures are due to SQLAlchemy's warning on which optuna depends.

```
E   sqlalchemy.exc.MovedIn20Warning: Deprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. To prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". Set environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
```

**Comment by takagi** [comment]
Blocked by #7276.

File: cupyx/scipy/interpolate/_bspline2.py [hunk 2]
```diff
@@ -530,7 +530,7 @@ def _make_periodic_spline(x, y, t, k, axis):
 
     # find the intervals for x[0] and x[-1]
     x0 = cupy.r_[x[0], x[-1]]
-    intervals_bc = cupy.array([k, n + k - 1], dtype=int)   # match scipy
+    intervals_bc = cupy.array([k, n + k - 1], dtype=cupy.int64)   # match scipy
```
Comment (kmaehashi) [inline cupyx/scipy/interpolate/_bspline2.py:533]: #7286 7280

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**